### PR TITLE
Manually mount serviceAccount token

### LINKED ISSUE #13108: ability to set automountServiceAccountToken to false
### What problem are you trying to solve?

To Meet Security Standards for not automatically mounting service accounts
for be compliance in Gov Clouds 

### How should the problem be solved?

Helm Chart should by default, set automountServiceAccountToken to true for all Linkerd Pods (to not change the current behavior)
but will have the option to change it to "false"

### Any alternatives you've considered?

None; its part of compliance tasks 
https://learn.microsoft.com/en-us/azure/governance/policy/samples/built-in-policies
https://www.azadvertizer.net/azpolicyadvertizer/kubernetes_block-automount-token.html

### How would users interact with this feature?

Set it to true in the helm chart for LInkerd in order to harden the security of the cluster to

### Would you like to work on this feature?

maybe

### PR DESCRIPTION
**Pull request by Aransh**
Subject
Disables "automountServiceAccountToken", instead manually mounts it as a projected volume where necessary

Problem
By default, kubernetes enables "automountServiceAccountToken" for all pods.
This poses a security risk, as pods might get kube-api permissions unintentionally.
More specifically, this fails security compliance tests:
https://learn.microsoft.com/en-us/azure/governance/policy/samples/built-in-policies
https://www.azadvertizer.net/azpolicyadvertizer/kubernetes_block-automount-token.html

Solution
Disable  "automountServiceAccountToken", create projected volume for the token, and mount it on relevant containers

Validation
Linkerd pods are able to access k8s API, work as expected (same as before)

Fixes #13108

### CODE REVIEW

**Review by malki-akamai — APPROVED** [review]

**Review by oyorgen — APPROVED** [review]

File: charts/partials/templates/_volumes.tpl [hunk 1]
```diff
@@ -18,3 +18,24 @@ projected:
       expirationSeconds: 86400 {{- /* # 24 hours */}}
       audience: identity.l5d.io
 {{- end -}}
+
+{{- define "partials.volumes.manualMountServiceAccountToken" -}}
```
Comment (alpeb) [inline charts/partials/templates/_volumes.tpl:22]: Can you rename this to make it consistent with the rest of the file?
```suggestion
{{- define "partials.volumes.manual-mount-service-account-token" -}}
```

**Review by alpeb — COMMENTED** [review]
Thanks @Aransh, this makes sense to me :+1: 
Could you also implement this for the viz, multicluster and jaeger extensions charts? Note that not all workloads there need to connect to the kube-api so mounting wouldn't always be necessary. Let me know if you need more assistance to figure that out.
Don't worry about the linkerd2-cni chart though, as there currently are other concerns that we need to address (linkerd/linkerd2-proxy-init#416).

**Comment by Aransh** [comment]
@alpeb I can gladly add that, but I don't currently use any of them, so would be difficult to figure out what’s required...
Could you help with pointing out which containers need kube-api access and which don't?

Also I need to see if I can figure out what's causing the CI to fail currently, will try and take a look next week

**Comment by Aransh** [comment]
Thanks for the CI clarification 😃

When you say "workload", you mean pod essentially? Keep in mind the token mounting is done per container, or do these workloads all have a single container?

If not/you’re not sure, I can just mount it to all containers, I'd just prefer to do it as precise as I can
For the linkers-control-plane for instance non of the init containers required it, and some of the containers too

**Comment by alpeb** [comment]
Yeah these are all single-container pods (not counting the proxy and init containers, which don't require any changes).

**Comment by Aransh** [comment]
Got it, will mount on all containers except unit and proxy then, probably will handle Tuesday.
Thanks!

File: charts/partials/templates/_volumes.tpl (hunk 1 above)
Comment (Aransh) [reply charts/partials/templates/_volumes.tpl:22]: Changed

**Comment by Aransh** [comment]
@alpeb Updated for linkerd-viz, linkerd-jaeger, linkerd-multicluster, and linkerd-multicluster-link as you requested.
I want to mention again I have not tested these as I don't use these charts, but they should see no difference as they all have the token manually mounted, except for the "linkerd-multicluster" gateway workload

**Comment by Aransh** [comment]
I see it's failing for mismatch with templates, but running `go test ./cli/cmd/... --update` as mentioned in docs doesn't seem to change anything after my latest changes... Am I missing something?

**Comment by alpeb** [comment]
Turns out those docs are outdated... the right command is `go test ./... -update`

**Review by alpeb — COMMENTED** [review]
This looks good to me, pending the remaining golden files updates :+1:

**Comment by Aransh** [comment]
@alpeb updated

**Review by alpeb — APPROVED** [review]

**Pull request merged**